FreeRTOS/Test/CMock/smp/single_priority_no_timeslice/covg_single_priority_no_timeslice_utest.c

```diff
@@ -415,3 +415,109 @@ void test_task_step_tick_xNextTaskUnblockTime_not_equal( void )
     xTickCount = 0U;
     vTaskStepTick((TickType_t)10U);
 }
+
+/**
+ * @brief xTaskResumeFromISR - resume higher priority suspended task
+ *
+ * This test resume a higher priority task from ISR when scheduler suspended. The
+ * return value of xTaskResumeFromISR indicates yield required for the core calling
+ * this API.
+ *
+ * <b>Coverage</b>
+ * @code{c}
+ * prvYieldForTask( pxTCB );
+ *
+ * if( xYieldPendings[ portGET_CORE_ID() ] != pdFALSE )
+ * {
+ *     xYieldRequired = pdTRUE;
+ * }
+ * @endcode
+ * ( xYieldPendings[ portGET_CORE_ID() ] != pdFALSE ) is true.
+ */
+void test_coverage_xTaskResumeFromISR_resume_higher_priority_suspended_task( void )
+{
+    TCB_t xTaskTCBs[ configNUMBER_OF_CORES + 1U ] = { NULL };
+    uint32_t i;
+    BaseType_t xReturn;
+
+    /* Setup the variables and structure. */
+    vListInitialise( &xSuspendedTaskList );
+    vListInitialise( &xPendingReadyList );
+    for( i = 0; i < configNUMBER_OF_CORES; i++ )
+    {
+        xTaskTCBs[ i ].uxPriority = 1;
+        xTaskTCBs[ i ].xTaskRunState = i;
+        xYieldPendings[ i ] = pdFALSE;
+        pxCurrentTCBs[ i ] = &xTaskTCBs[ i ];
+    }
+    /* A suspended task is created to be resumed from ISR. The task has higher priority
+     * than uxTopReadyPriority and the scheduler is suspended. The task will be added
+     * to xPendingReadyList after resumed from ISR. */
+    xTaskTCBs[ configNUMBER_OF_CORES ].uxPriority = 2;
+    listINSERT_END( &xSuspendedTaskList, &xTaskTCBs[ i ].xStateListItem );
+    uxTopReadyPriority = 1;
+    uxSchedulerSuspended = pdTRUE;
+
+    /* Expections. */
+    vFakePortAssertIfInterruptPriorityInvalid_Ignore();
+
+    /* API calls. */
+    xReturn = xTaskResumeFromISR( &xTaskTCBs[ i ] );
+
+    /* Validateions. In single priority test, the calling core is requested to yield
+     * since a higher priority task is resumed. */
+    TEST_ASSERT( xReturn == pdTRUE );
+}
+
+/**
+ * @brief xTaskResumeFromISR - resume lower priority suspended task
+ *
+ * This test resume a lower priority task from ISR when scheduler suspended. The
+ * return value of xTaskResumeFromISR indicates yield not required for the core
+ * calling this API.
+ *
+ * <b>Coverage</b>
+ * @code{c}
+ * prvYieldForTask( pxTCB );
```

Comment:
xTaskResumeFromISR() ?

Reply:
This is the code snippet in xTaskResumeFromISR().
I will add more code snippet to make it more clear to the readers.
https://github.com/chinglee-iot/FreeRTOS-Kernel/blob/b266846c84a18491a3f186d76a71eacacfe567d8/tasks.c#L3014-L3023